packages/datatrak-web-server/src/routes/SurveyRoute.ts

```diff
@@ -78,13 +79,24 @@ export class SurveyRoute extends Route<SurveyRequest> {
       ctx,
       query = {},
       params: { surveyCode },
+      models,
     } = this.req;
     const { fields = DEFAULT_FIELDS } = query;
+    // check if survey exists in the database
+    const dbSurveyResults = await models.survey.find({ code: surveyCode });
+
+    if (!dbSurveyResults.length) throw new Error(`Survey with code ${surveyCode} not found`);
```

Comment:
There's an interesting discussion to be had about whether a server should return different errors for resource does not exist vs don't have permissions to the resource. Some people advocate to not differentiate the two since it gives external actors information about the system that they might not need to know about.

I feel the risks of doing so probably aren't that serious though, and it can provide a better user experience as we're doing here. That being said, I like that it keeps the code a little simpler if the two cases are treated the same.

Just some meandering early morning thoughts here :relaxed: